### PR TITLE
JAMES-3693 Update redis config to support master-replica topology

### PR DESCRIPTION
**Pull request by hungphan227**
https://github.com/linagora/james-project/issues/5171

### CODE REVIEW

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/redis.adoc [hunk 1]
```diff
@@ -15,8 +15,8 @@ to get some examples and hints.
 | redisURL
 | the Redis URI pointing to Redis server. Compulsory.
 
-| cluster.enabled
-| Whether we should attempt connections to Redis as a cluster.Defaults to false.
+| redis.topology
+| Redis server topology. Defaults to standalone. Possible values: standalone, cluster, master-replica
```
Comment (quantranhong1999) [inline server/apps/distributed-app/docs/modules/ROOT/pages/configure/redis.adoc:19]: This is a breaking change.

IMO we should either provide retro-compatibility for `cluster.enabled`, or docs the breaking changes in `upgrade-instructions.md`.

**Review by quantranhong1999 — COMMENTED** [review]
Looks good to me overall.

File: server/mailet/rate-limiter-redis/src/test/java/org/apache/james/rate/limiter/RedisRateLimiterWithMasterReplicaTopologyTest.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package org.apache.james.rate.limiter
```
Comment (Arsnael) [inline server/mailet/rate-limiter-redis/src/test/java/org/apache/james/rate/limiter/RedisRateLimiterWithMasterReplicaTopologyTest.scala:1]: Missing license :)

**Review by vttranlina — COMMENTED** [review]
LGTM

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/redis.adoc (hunk 1 above)
Comment (hungphan227) [reply server/apps/distributed-app/docs/modules/ROOT/pages/configure/redis.adoc:19]: updated upgrade-instructions.md

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**